### ISSUE TITLE
getting existence_error(procedure,http_stream:close_range/3) with rdf_read_turtle/3

### ISSUE THREAD

**Issue by josd** [SWI-Prolog/swipl-devel#1239]
With the latest swipl version 9.3.1-1-g16d16ef5f we are getting `existence_error(procedure,http_stream:close_range/3)` with `rdf_read_turtle/3`
```
$ eye --quiet --turtle http://eyereasoner.github.io/eye/reasoning/bmt/1tt8.ttl
** ERROR ** gre ** error(existence_error(procedure,http_stream:close_range/3),context(system:call/3,_14406))
```
When we comment https://github.com/eyereasoner/eye/blob/3bf91d1335754170e92e11b3ddf42b57df5e71c6/eye.pl#L1216 we still get a warning
```
$ eye --quiet --turtle http://eyereasoner.github.io/eye/reasoning/bmt/1tt8.ttl
Warning: call/3: Unknown procedure: http_stream:close_range/3
```

**Comment by josd** [comment]
The error and warning are gone with
```
$ git diff .
diff --git a/http_open.pl b/http_open.pl
index df72832..375bba6 100644
--- a/http_open.pl
+++ b/http_open.pl
@@ -1584,8 +1584,7 @@ consider_keep_alive(Lines, Parts, _Host, StreamPair, In, _Options) :-
     !,
     stream_pair(StreamPair, In0, _),
     stream_range_open(In0, In,
-                      [ size(Bytes),
-                        onclose(close_range(StreamPair))
+                      [ size(Bytes)
                       ]).
 consider_keep_alive(_, _, _, Stream, Stream, _).
```
but this is of course a crude workaround.

**Comment by josd** [comment]
I guess this might be better
```
$ git diff
diff --git a/http_open.pl b/http_open.pl
index df72832..c058a40 100644
--- a/http_open.pl
+++ b/http_open.pl
@@ -1585,7 +1585,7 @@ consider_keep_alive(Lines, Parts, _Host, StreamPair, In, _Options) :-
     stream_pair(StreamPair, In0, _),
     stream_range_open(In0, In,
                       [ size(Bytes),
-                        onclose(close_range(StreamPair))
+                        onclose(http_open:close_range(StreamPair))
                       ]).
 consider_keep_alive(_, _, _, Stream, Stream, _).
```

**Comment by jan-swi-prolog-01** [comment]
This problem should be gone with the 9.3.2 release (actually quite a bit earlier).

**Comment by josd** [comment]
We are using SWI-Prolog (threaded, 64 bits, version 9.3.2-2-ge4d5cf64e) and still have the https://github.com/SWI-Prolog/swipl-devel/issues/1239#issue-2138613939 problem.
Now also did a fresh git clone and a fresh build with https://github.com/josd/josd.github.io/blob/master/temp/mkswipl but still the same.
What we basically do is
```
:- use_module(library(http/http_open)).
...
    http_open(Arg, In, []),
    set_stream(In, encoding(utf8)),
...
    rdf_read_turtle(stream(In), Triples, [on_error(error)]),
    close(In),
```
and the `close(In)` causes the error `error(existence_error(procedure,http_stream:close_range/3),context(system:call/3,_11806))`

**Comment by jan-swi-prolog-01** [comment]
Odd.   Sure you properly updated the submodules?  The patch is in `packages/http`, commit 04d739dda642cac645ce6a4f4b4999e2ce2e5643.    What platform are we talking about?

**Comment by josd** [comment]
package/http is at
```
~/github.com/SWI-Prolog/swipl-devel/packages/http$ git log
commit 857b5e7234f02d91904a485d1417f1e85c0e18cc (HEAD, tag: V9.3.2, tag: V9.2.2, origin/master, origin/HEAD, master)
Author: Jan Wielemaker <jan@swi-prolog.org>
Date:   Thu Feb 29 09:55:45 2024 +0100

    Avoid unused function warning
```
and the platform is
```
$ uname -srmo
Linux 5.15.133.1-microsoft-standard-WSL2 x86_64 GNU/Linux
```
```
$ cat /etc/issue
Ubuntu 22.04.4 LTS \n \l
```

**Comment by JanWielemaker** [comment]
Very odd.    Well, it reproduces, so I'll figure out what is wrong.    First some other work though ...

**Comment by josd** [comment]
Thanks in advance Jan 👍

**Comment by JanWielemaker** [comment]
Fixed with eac35cebf6ac07c156ce4b35f2a5eaa66537888a.   The meta-predicate property of `stream_range_open/3` was lost in the saved state :cry:

**Comment by josd** [comment]
Thanks Jan, it works fine now!
